node_modules/oae-folders/lib/api.js

```diff
@@ -928,70 +943,83 @@ var setFolderPermissions = module.exports.setFolderPermissions = function(ctx, f
-                                'principalIds': newMemberIds
-                            }, 'An error occurred while adding folder to member libraries after role assignment');
+                            return callback(err);
                         }
 
-                        // Remove the folder from the libraries of removed members
-                        FoldersFoldersLibrary.remove(removedMemberIds, folder, function(err) {
+                        // Insert the folder into the libraries of new members
+                        FoldersFoldersLibrary.insert(newMemberIds, folder, function(err) {
                             if (err) {
                                 log(ctx).warn({
                                     'err': err,
                                     'folderId': folder.id,
-                                    'principalIds': removedMemberIds
-                                }, 'An error occurred while removing folder from member libraries after role removal');
+                                    'principalIds': newMemberIds
+                                }, 'An error occurred while adding folder to member libraries after role assignment');
                             }
 
-                            // TODO: Test timestamp doesn't update with subsequent set-permissions
-                            // For all current members, update the folder in their libraries. This includes
-                            // members that were just added which is a bit of a waste, but easier to code
-                            var updateMemberIds = _.keys(membershipAfterChanges);
-                            OaeUtil.invokeIfNecessary(_testLibraryUpdateThreshold(folder), FoldersFoldersLibrary.update, updateMemberIds, folder, null, function(err) {
+                            // Remove the folder from the libraries of removed members
+                            FoldersFoldersLibrary.remove(removedMemberIds, folder, function(err) {
                                 if (err) {
                                     log(ctx).warn({
                                         'err': err,
                                         'folderId': folder.id,
-                                        'principalIds': updateMemberIds
-                                    }, 'An error occurred while updating folder in member libraries after having permissions updated');
+                                        'principalIds': removedMemberIds
+                                    }, 'An error occurred while removing folder from member libraries after role removal');
                                 }
 
-                                FoldersAPI.emit(FoldersConstants.events.UPDATED_FOLDER_MEMBERS, ctx, folder, _.pick(permissionChanges, updatedMemberIds), newMemberIds, updatedMemberIds, removedMemberIds);
-                                return callback();
+                                // TODO: Test timestamp doesn't update with subsequent set-permissions
```

Comment:
Yes, it's just asking for a test that does:
1. Create a set of folders
2. Set some permissions on folder A
3. let timestampA  = folderA.lastUpdated (should be nowish)
4. Set some more permissions on folder A
5. assert folderA.lastUpdated is still the same as timstampA as we don't update the folder libraries if the last update was less than an hour ago.

This can be outscoped for now